pkg/controller/utils/datadog/metrics_forwarder_test.go

```diff
@@ -624,3 +635,184 @@ func TestMetricsForwarder_setTags(t *testing.T) {
 		})
 	}
 }
+
+func Test_metricsForwarder_processReconcileError(t *testing.T) {
+	nsn := types.NamespacedName{
+		Namespace: "foo",
+		Name:      "bar",
+	}
+	mf := &metricsForwarder{
+		namespacedName: nsn,
+	}
+	mf.initGlobalTags()
+
+	tests := []struct {
+		name     string
+		loadFunc func() (*metricsForwarder, *fakeMetricsForwarder)
+		err      error
+		wantErr  bool
+		wantFunc func(*fakeMetricsForwarder) error
+	}{
+		{
+			name: "last error init value, new unknown error => send unsucess metric",
+			loadFunc: func() (*metricsForwarder, *fakeMetricsForwarder) {
+				f := &fakeMetricsForwarder{}
+				f.On("delegatedSendReconcileMetric", 0.0, []string{"cr_namespace:foo", "cr_name:bar", "reconcile_err:err_msg"})
+				mf.delegator = f
+				mf.lastReconcileErr = errInitValue
+				return mf, f
+			},
+			err:     errors.New("err_msg"),
+			wantErr: false,
+			wantFunc: func(f *fakeMetricsForwarder) error {
+				if !f.AssertCalled(t, "delegatedSendReconcileMetric", 0.0, []string{"cr_namespace:foo", "cr_name:bar", "reconcile_err:err_msg"}) {
+					return errors.New("Function not called")
+				}
+				if !f.AssertNumberOfCalls(t, "delegatedSendReconcileMetric", 1) {
+					return errors.New("Wrong number of calls")
+				}
+				return nil
+			},
+		},
+		{
+			name: "last error init value, new auth error => send unsucess metric",
+			loadFunc: func() (*metricsForwarder, *fakeMetricsForwarder) {
+				f := &fakeMetricsForwarder{}
+				f.On("delegatedSendReconcileMetric", 0.0, []string{"cr_namespace:foo", "cr_name:bar", "reconcile_err:Unauthorized"})
```

Comment:
You could add `.Once()` here then below instead of `AssertCalled` and `AssertNumberOfCalls` you could do `AssertExpectations(t)`. It might make sense to do this for cases when you expect no calls as well.

Reply:
thanks 💯